pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -502,6 +603,40 @@ func buildStatefulSetModificationFunction(mdb mdbv1.MongoDB) statefulset.Modific
 
 	dataVolume := statefulset.CreateVolumeMount(dataVolumeName, "/data")
 
+	agentVolumeMounts := []corev1.VolumeMount{agentHealthStatusVolumeMount, automationConfigVolumeMount, dataVolume}
+	mongodVolumeMounts := []corev1.VolumeMount{mongodHealthStatusVolumeMount, hooksVolumeMount, dataVolume}
+
+	tlsPodSpec := podtemplatespec.Apply()
```

Comment:
there should be a `podtemplatespec.NOOP()` that is semantically more correct for what you need here

Reply:
Didn't know about that, thanks!